test-loop-tests/src/utils/peer_manager_actor.rs

```diff
@@ -170,23 +240,55 @@ impl TestLoopNetworkSharedState {
                 tx_processor_sender: TxRequestHandleSenderForTestLoopNetwork::from(data),
                 partial_witness_sender: PartialWitnessSenderForNetwork::from(data),
                 shards_manager_sender: Sender::<ShardsManagerRequestFromNetwork>::from(data),
+                peer_manager_sender: Sender::<TestLoopNetworkBlockInfo>::from(data),
             }),
         );
     }
 
+    /// Stops processing of requests from `from` peer to `to` peer.
+    pub fn disallow_requests(&self, from: PeerId, to: PeerId) {
+        let mut guard = self.0.lock().unwrap();
+        guard.disallowed_peer_links.entry(from).or_default().insert(to);
+    }
+
+    /// Allows processing of requests between all peers.
+    pub fn allow_all_requests(&self) {
+        let mut guard = self.0.lock().unwrap();
+        guard.disallowed_peer_links = HashMap::new();
+    }
+
     fn account_to_peer_id(&self, account_id: &AccountId) -> PeerId {
         let guard = self.0.lock().unwrap();
         guard.account_to_peer_id.get(account_id).unwrap().clone()
     }
 
-    fn senders_for_account(&self, account_id: &AccountId) -> Arc<OneClientSenders> {
+    fn is_peer_link_disallowed(
+        guard: &MutexGuard<TestLoopNetworkSharedStateInner>,
+        from: &PeerId,
+        to: &PeerId,
+    ) -> bool {
+        guard.disallowed_peer_links.get(from).and_then(|blocklist| blocklist.get(to)).is_some()
+    }
+    fn senders_for_account(
+        &self,
+        origin: &AccountId,
+        account_id: &AccountId,
+    ) -> Option<Arc<OneClientSenders>> {
```

Comment:
Instead of returning `Option<Arc<OneClientSenders>>`, for the case of disallowed link, would it be better to return a "black hole client sender" that basically has a bunch of `noop()` senders + maybe some logging that the message is dropped?

We can even make it a test loop sender with some specific identifier that we can disallow using `remove_events_with_identifier`. Would be much better for tracking.

That'll also minimize the changes to the rest of the default handlers.

Reply:
Good idea, done in e9962284e6fc14c81f901fd0acec603f10103497